app/models/miq_queue_worker_base/runner.rb

```diff
@@ -143,6 +145,21 @@ def do_work
 
   private
 
+  def register_worker_with_worker_monitor
+    worker_monitor_drb.register_worker(@worker.pid, @worker.class.name, @worker.queue_name)
+  rescue DRb::DRbError => err
+    do_exit("Failed to register worker with worker monitor: #{err.class.name}: #{err.message}", 1)
+  end
+
+  def drb_dequeue_available?
+    @drb_dequeue_available ||=
+      begin
+        server.drb_uri.present? && worker_monitor_drb.respond_to?(:register_worker)
+      rescue DRb::DRbError
```

Comment:
Was there something else in this block before?  How would DRb::DRbError be raised?  We're not doing anything with drb, right?

Reply:
ah, I see, we used to call `worker_monitor_drb.register_worker` which could raise that exception...

Reply:
Right, so in this case I just need some call to the drb object to raise to tell if it's available or not. I used a `respond_to?` here because that proxies over to the server side without actually doing anything.

Reply:
Exactly.  I don't think the begin/rescue is needed here.

Reply:
(it's needed in the method above that calls `register_worker` on the drb object)

Reply:
This is very much needed I think. `respond_to?` is proxied to the remote drb object so it will raise rather than return false if the drb server is inaccessible.

I would love a way to tell if the drb server is up that didn't rely on exception handling, do you know of something like that?

Reply:
@agrare do you know if there's anything like that?

Reply:
My bad.  I forgot the drb object was proxied and could raise here.  Oh DRb.

Reply:
Not that I'm aware of, in `miq_fault_tolerant_vim` which wraps `MiqVim` to deal with DRb "issues" we just catch `DRb::DRbConnError` and retry if the broker is on a new port